### DISCUSSION TITLE
linkToInput — Channel inner not supported in function isField

### DISCUSSION THREAD

**Discussion by tvideo** [totumonline/totum-ru-issues-and-discussions#121] [Как это сделать?] [answered]
Здравствуйте. 
Подскажите плиз по вопросику:
Есть простая таблица, в ней две кнопки с вызовом linkToInput. При вводе цифры - изменяется количество. Работало, а потом начала выдавать ошибку "Channel inner not supported in function isField"
Код в кнопке:

```
= : linkToInput(html: "Введите количество"; title: "Изменить количество"; code: $stradd; type: "number"; refresh: true; button: "Добавить")
stradd: strAdd(str: '=: set(table: $#ntn; field: "kolvo" = $set1; where: "id" = #id)'; str: $#nl; str: 'set1: #kolvo + $#input')
```

![1](https://user-images.githubusercontent.com/32614108/165166183-2b90d495-70fe-4448-ba25-b11e6bd4f7b6.png)
![2](https://user-images.githubusercontent.com/32614108/165166201-881076b0-0e31-4529-a16b-a5d8a76dc2e0.png)
![3](https://user-images.githubusercontent.com/32614108/165166211-a8b69641-d4d3-468b-9d66-a572e36636b1.png)

В чем может быть причина?
Спасибо

**Comment by ttm-alexeypolunin** [comment]
Должно работать, а какая версия? Я проверил на текущей и типа ок.

Может там еще какие-то действия происходят после изменения количества?

**Reply by tvideo** [reply]
Больше действий нет, код кнопки:
= : linkToInput(html: "Введите количество"; title: "Изменить количество"; code: $stradd; type: "number"; refresh: true; button: "Добавить")
stradd: strAdd(str: '=: set(table: $#ntn; field: "kolvo" = $set1; where: "id" = #id)'; str: $#nl; str: 'set1: #kolvo + $#input')

Самое интересное что все работало) Потом я добавил префильтр и все.
Версия v1.0.24

**Reply by tvideo** [reply]
Сделал тестовую таблицу. Без префильтра все ок, с префильтром - ошибка. При этом ошибка появляется даже если префильтр не ссылается на поле "kolvo". 
Ответ должен быть где-то в префильтре)

![2022-04-26_07-53-37](https://user-images.githubusercontent.com/32614108/165224421-b5fef731-78b6-4323-9f25-b4806dfe7e58.png)
.

**Comment by poluninatm** [comment, marked as answer]
Добрый день. Актуальная версия Тотум сейчас 2.3.38.20. Префильтры мы переписывали, сейчас они эту ошибку уже не вызывают.

**Reply by tvideo** [reply]
Обновил до версии 2.3.38.20
"linkToInput" работает, но без префильтра.
Теперь следующая проблема:
Добавляю префильтр типа "Строка", все ок 

![2022-04-26_13-56-19](https://user-images.githubusercontent.com/32614108/165285408-1d26b8eb-43bb-49ed-ade0-be53f3f6dfc5.png)

Как только его тип меняю на "Селект-дерево", сразу уходим в error:

![2022-04-26_13-58-37](https://user-images.githubusercontent.com/32614108/165285637-aff51dd5-eec9-4c67-ab19-b0fd6b626574.png)

**Reply by tvideo** [reply]
Даже если просто создать простую таблицу и просто создать префильтр с типом "Селект-дерево", сразу ухожу в ошибку, которая выше. В чем может быть причина?

**Reply by poluninatm** [reply]
Поправим этот warning  в ближайшем релизе. Пока можно поставить галочку  "Значение по умолчанию" и он перестанет выпадать

**Reply by tvideo** [reply]
"Значение по умолчанию" помогло. 
Спасибо
